ddtrace/appsec/_remoteconfiguration.py

```diff
@@ -60,7 +61,7 @@ def enable_appsec_rc(test_tracer: Optional[Tracer] = None) -> None:
         or AppSecRC(_preprocess_results_appsec_1click_activation, _appsec_callback)
     )
 
-    if _appsec_rc_features_is_enabled():
+    if _asm_feature_is_required():
```

Comment:
Is this function correct? If _appsec_enabled is enabled, BUT config._api_security_enabled is not, we don't register ASM_FEATURES.

The previous `_appsec_rc_features_is_enabled` function doesn't validate if Appsec is enabled or not. That function checks if DD_APPSEC_ENABLED env var is present.

```
def _appsec_rc_features_is_enabled() -> bool:
    if config._remote_config_enabled:
        return APPSEC_ENV not in os.environ
    return False
```

Reply:
I think it is. 
`return Flags.ASM_ACTIVATION in flags or Flags.ASM_API_SECURITY_SAMPLE_RATE in flags`
So either ASM_ACTIVATION flag is present (equivalent to _appsec_rc_features_is_enabled True) OR api security is activated.

Reply:
ups, sorry, I mixed the functions `_asm_feature_is_required` and `_appsec_apisec_features_is_active` :+1: